vscode/src/analysis/runAnalysis.ts

```diff
@@ -15,6 +16,10 @@ export const registerAnalysisTrigger = (
         return;
       }
 
+      if (isUriIgnored(e.document.uri)) {
```

Comment:
we can get rid of the notifyFileChanges in ChangeTextDocument event altogether since we force the user to save the file on agent changes.